src/Microsoft.TestPlatform.CommunicationUtilities/SocketClient.cs

```diff
@@ -69,7 +70,7 @@ private void OnServerConnected(Task connectAsyncTask)
                 throw connectAsyncTask.Exception;
             }
 
-            this.channel = this.channelFactory(this.tcpClient.GetStream());
+            this.channel = this.channelFactory(new PlatformStream().PlatformBufferedStream(this.tcpClient.GetStream(), SocketConstants.BUFFERSIZE));
```

Comment:
Seems like we are using `BufferedStream` for both Read and Write operations. We need to change it to comply with recommended approach.

Reply:
Same applies for SocketServer.cs